e2e/test_search_immunization.py

```diff
@@ -73,3 +73,16 @@ def test_search_ignore_deleted(self):
         self.assertTrue(mmr1["id"] in resource_ids)
         self.assertTrue(mmr2["id"] in resource_ids)
         self.assertTrue(deleted_mmr["id"] not in resource_ids)
+
+    def test_search_reject_tbc(self):
+        # Given patient has a vaccine with no NHS number
+        imms = create_an_imms_obj(str(uuid.uuid4()), "TBC", mmr_code)
+        del imms["contained"][1]["identifier"][0]["value"]
+        imms["contained"][1]["identifier"][0]["extension"][0]["valueCodeableConcept"]["coding"][0]["code"] = "04"
+        self.store_records(imms)
+
+        # When
+        response = self.default_imms_api.search_immunizations("TBC", "MMR")
+
+        # Then
+        self.assertEqual(response.status_code, 500, response.text)
```

Comment:
Why 500? It should be 400? or 400 range.

Reply:
That's what I thought, but `InvalidPatientId` error class is set to use `Code.server_error`. Do you want me to change it as part of this ticket?
![image](https://github.com/NHSDigital/immunisation-fhir-api/assets/35727498/339846bb-890a-4c58-a54b-0bfe8d79a768)

Reply:
Yes please. That should be a mistake. I can't think of any scenario that causes a server error. If we know nhs-number is wrong then it's a bad request.

Reply:
This test no longer exists but I've changed the code in the error class to `invariant` from `server_error`